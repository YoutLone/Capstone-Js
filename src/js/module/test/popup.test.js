import generatePopupContent from '../popup/popupWindow.js';

jest.mock('../popup/commentshow.js', () => ({
  __esModule: true,
  default: jest.fn(
    (comment) => `<div class="comment">Comment ${comment.id}</div>`,
  ),
}));

test('generatePopupContent should generate correct popup content', () => {
  const data = {
    image: { medium: 'image.jpg' },
    name: 'John Doe',
    gender: 'Male',
    country: { name: 'Country', timezone: 'Timezone' },
  };

  const comments = [
    { id: 1, username: 'user1', comment: 'Comment 1' },
    { id: 2, username: 'user2', comment: 'Comment 2' },
    { id: 3, username: 'user3', comment: 'Comment 3' },
  ];

  const generatedHTML = generatePopupContent(data, comments);

  expect(generatedHTML).toMatchSnapshot();
});
