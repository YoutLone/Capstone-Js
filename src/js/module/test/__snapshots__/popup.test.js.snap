// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generatePopupContent should generate correct popup content 1`] = `
"
    <div class="popup-content container">
      <i class="fa-solid fa-xmark close-btn"></i>
      <img src="image.jpg" alt="#" class="popup-image" />
      <p class="popup-name">Name: John Doe</p>
      <p class="popup-gender">Gender: Male</p>
      <p class="popup-country">Country: Country</p>
      <p class="popup-timezone">Timezone: Timezone</p>
    </div>
    <h2>Comments (3)</h2>
    <div class="comment">Comment 1</div><div class="comment">Comment 2</div><div class="comment">Comment 3</div>
    
    <h2>Add a comment</h2>
    <form id="comment-form">
      <label for="name-input">Your name</label>
      <input type="text" id="name-input" placeholder="Enter your name" required>
      <label for="insights-input">Your insights</label>
      <input type="text" id="insights-input" placeholder="Enter your insights" required>
      <button id="comment-button" type="submit">Comment</button>
    </form>
  
  "
`;
